pkg/operator/ceph/file/mds/spec.go

```diff
@@ -150,3 +170,31 @@ func deleteMdsDeployment(context *clusterd.Context, namespace string, deployment
 	}
 	return nil
 }
+
+func appendZoneAntiAffinity(pod *v1.PodSpec) {
+	// Create the antiAffinity for the MDS pod
+	mdsZoneAntiAffinity := v1.WeightedPodAffinityTerm{
+		Weight: 10,
+		PodAffinityTerm: v1.PodAffinityTerm{
+			LabelSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      k8sutil.AppAttr,
+						Operator: metav1.LabelSelectorOpIn,
+						Values:   []string{"rook-ceph-mds"},
+					},
+				},
+			},
+			TopologyKey: labelZoneFailureDomain,
+		},
+	}
+	// Ensure that pod.Affinity is non-nil
+	if pod.Affinity.PodAntiAffinity == nil {
+		pod.Affinity.PodAntiAffinity = &v1.PodAntiAffinity{}
+	}
+	paa := pod.Affinity.PodAntiAffinity
```

Comment:
If the admin has specified anti-affinity in the filesystem CR, I don't think we should append the default anti-affinity.